### PR TITLE
Start testing under Python 3

### PR DESCRIPTION
**Pull request by jmbowman**
We want to start running the `make upgrade` Jenkins job under Python 3 for some repos, which means the code here needs to work under it.

### CODE REVIEW

File: jenkins/edx_platform_test_notifier.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
+from __future__ import absolute_import
+
 import logging
-import os
```
Comment (estute) [inline jenkins/edx_platform_test_notifier.py]: ^ I notice you removed this as an import but not from the code. Was this just an unused import?

File: jenkins/tests/test_timeout.py [hunk 2]
```diff
@@ -63,7 +65,7 @@ def test_stop_stuck_builds_with_stuck(self, mock_desc,
 
         stop_build.assert_has_calls([call(3), call(2)], any_order=True)
 
-        update_desc.assert_hass_calls(
```
Comment (estute) [inline jenkins/tests/test_timeout.py]: How did this pass before this?!

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+; D406 = Section name should end with a newline (numpy style)
+; D407 = Missing dashed underline after section (numpy style)
+; D408 = Section underline should be in the line following the section’s name (numpy style)
+; D409 = Section underline should match the length of its name (numpy style)
+; D410 = Missing blank line after section (numpy style)
+; D411 = Missing blank line before section (numpy style)
+; D412 = No blank lines allowed between a section header and its content (numpy style)
+; D413 = Missing blank line after last section (numpy style)
+; D414 = Section has no content (numpy style)
+ignore = D101,D200,D203,D212,D215,D404,D405,D406,D407,D408,D409,D410,D411,D412,D413,D414
```
Comment (estute) [inline tox.ini]: Why two `pycodestyle` blocks?

File: aws/deregister_amis.py [hunk 4]
```diff
@@ -66,18 +66,6 @@ def main(raw_args):
...
-        dest='aws_access_key_id',
-        help='aws access key id',
-        default=os.environ.get("AWS_ACCESS_KEY_ID"),
-        )
-    parser.add_argument(
-        '--aws-secret-access-key', '-s',
-        dest='aws_secret_access_key',
-        help='aws secret access key',
-        default=os.environ.get("AWS_SECRET_ACCESS_KEY"),
-    )
```
Comment (estute) [inline aws/deregister_amis.py]: ^ won't we still need these? Or is the fact that the jenkins worker has an IAM role what allows this script to function w/o credentials?

File: aws/tests/test_deregister_amis.py [hunk 5]
```diff
@@ -1,47 +1,63 @@
...
-        return self.conn.get_all_images()[0]
-
-    def test_main(self):
+    @patch('boto3.resource', return_value=MockEC2(matching_image_exists=False))
+    def test_main(self, mock_ec2):
         """
         Test output of main
         """
-        with LogCapture() as l:
+        with LogCapture() as capture:
```
Comment (estute) [inline aws/tests/test_deregister_amis.py]: wow, thank you for changing this variable name! ;)

File: aws/deregister_amis.py (hunk 4 above)
Comment (jmbowman) [reply aws/deregister_amis.py]: They're picked up from the environment variables.  boto3 dropped easy support for passing these in explicitly via the API in order to discourage people from listing the credentials in their source code.  And in practice, our Jenkins job wasn't using these command line parameters.

File: aws/tests/test_deregister_amis.py (hunk 5 above)
Comment (jmbowman) [reply aws/tests/test_deregister_amis.py]: The new pycodestyle complained about it...

File: jenkins/edx_platform_test_notifier.py (hunk 1 above)
Comment (jmbowman) [reply jenkins/edx_platform_test_notifier.py]: Yes, PyCharm caught that.

File: jenkins/tests/test_timeout.py (hunk 2 above)
Comment (jmbowman) [reply jenkins/tests/test_timeout.py]: The old version of mock just treated it like any other function call on a Mock and returned a new Mock (so it didn't actually do an assertion at all).  The new version is smart enough to realize "hmm, that looks like you were trying to call a function that actually exists..." and raise an error.

**Pull request merged**